algo/src/main/java/org/neo4j/graphalgo/impl/msbfs/BiMultiBitSet32.java

```diff
@@ -0,0 +1,139 @@
+package org.neo4j.graphalgo.impl.msbfs;
+
+import java.util.Arrays;
+
+/**
+ * Two 32-bit wide BitSets per node.
+ * This class represents the fixed-size bit field as described in [1].
+ * Where two bit sets are stored in the same cache line to improve locality
+ * on sequential access.
+ * It is equivalent to a {@code List<List<BitSet>>}, but implemented more
+ * efficiently as a sparse {@code long[]}.
+ * The {@code ω} parameter is fixed to 32, that is, only 32 sources can be
```

Comment:
@knutwalker oh right, because you access (therefore store) `next` & `seen` together? (in `NextAndSeenLocal`)